### PR TITLE
Add "source" filter

### PR DESCRIPTION
**Pull request by dongbohu**
This PR addresses #39. I have tested it on my local env.

### CODE REVIEW

**Comment by dongbohu** [comment]
@ravila4 I am not an expert on Tornado web handlers. Please feel free to offer any suggestions. Thanks.

File: src/config_web.py [hunk 1]
```diff
@@ -95,6 +95,35 @@
...
+    'source': {
+        'type': list,
+        'default': ['all'],
+        'max': 1000,
+        'group': 'esqb',
+        'translations': [
+            (re.compile(src, re.I), src) for src in SOURCES
+        ]
+    },
+    'source_facet_filter': {
```
Comment (dongbohu) [inline src/config_web.py:113]: @ravila4 What is this filter for? I copied from the `species` implementation.

File: src/config_web.py [hunk 2]
```diff
@@ -95,6 +95,35 @@
...
+]
+
+SOURCE_TYPEDEF = {
+    'source': {
+        'type': list,
+        'default': ['all'],
+        'max': 1000,
+        'group': 'esqb',
+        'translations': [
+            (re.compile(src, re.I), src) for src in SOURCES
```
[the remark this answers is not in the document]
Comment (ravila4) [reply src/config_web.py:110]: Same applies to the SOURCES variable. It is only used in the species filter to provide translations from taxid to species name.

**Review by ravila4 — COMMENTED** [review]
I tested the filters, and they are working as expected. Just a couple of comments on the code.

**Comment by dongbohu** [comment]
@ravila4 Thank you for your comments. I have removed those unnecessary lines based on your feedback. Please feel free to close it if it's okay now.

**Pull request merged**